picasso/src/main/java/com/squareup/picasso/MarkableInputStream.java

```diff
@@ -127,6 +131,9 @@ private void skip(long current, long target) throws IOException {
   }
 
   @Override public int read(byte[] buffer) throws IOException {
+    if (offset + buffer.length >= limit) {
+      setLimit(2 * (offset + buffer.length));
```

Comment:
Yeah. If we can detect that case and not change the limit then I think this will work.